spec/models/reports/facility_state_spec.rb

```diff
@@ -67,4 +59,103 @@
       end
     end
   end
+
+  context "assigned patients by care states" do
+    describe "under_care, lost_to_follow_up, dead" do
+      it "computes the number of assigned patients correctly with each care state" do
+        facility = create(:facility)
+        _dead_patient = create(:patient, recorded_at: june_2021[:long_ago], assigned_facility: facility, status: "dead")
+        _under_care_patient = create(:patient, recorded_at: june_2021[:under_12_months_ago], assigned_facility: facility)
+        ltfu_patient = create(:patient, recorded_at: june_2021[:over_12_months_ago], assigned_facility: facility)
+        create(:blood_pressure, patient: ltfu_patient, recorded_at: june_2021[:over_12_months_ago])
+        _different_facility_patient = create(:patient, recorded_at: june_2021[:long_ago])
+
+        RefreshMaterializedViews.new.refresh_v2
+        with_reporting_time_zone do
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).under_care).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).lost_to_follow_up).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).dead).to eq 1
```

Comment:
Can we make 1, 2, and 3, of these patients respectively? Keeping everything as `1` increases the risk of false positives, what do you think?

Reply:
Thanks for prodding this @harimohanraj89 🙇 Done in bc96f9f31bad66f3aead8dd904610c61192a47b7 and d45f60e5b808eece609704a01ebddb336a09716e.